app/js/actions/connection.js

```diff
@@ -48,12 +45,13 @@ const shakeHandsWithServer = () => {
     return (dispatch, getState) => {
         dispatch(updateConnectionStatus('Validating credentials'));
         const { connection: { uri, availableApis, secretKey } } = getState();
+        const url = `http://${uri.split('/')[0]}${availableApis[0]}`;
         const httpVerb = 'POST';
         const requestTime = Date.now();
         const body = JSON.stringify({});
-        const digest = makeB64Digest(secretKey, httpVerb, requestTime, body);
+        const digest = makeB64Digest(secretKey, httpVerb, url, requestTime, body);
```

Comment:
Actually, `body` is used in the `fetch` call. So in the interest of making the signature use the same content as the `fetch` I'm going to leave as is. We should refactor this code at some point.